### PR TITLE
doc/user: complete + polish release notes for v0.19.0

### PR DESCRIPTION
**Pull request by benesch**
### Motivation

* Weekly release notes PR.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR adds a release note for any [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/user/content/release-notes.md#what-changes-require-a-release-note).

### CODE REVIEW

**Review by morsapaes — APPROVED** [review]
We're including two experimental features, which goes against our own [guidelines](https://github.com/MaterializeInc/materialize/pull/9819#pullrequestreview-842259190). I'd remove these mentions, but double checking since I might just be out of the loop!

File: doc/user/content/release-notes.md [hunk 1]
```diff
@@ -107,35 +107,46 @@ These changes are present in [unstable builds](/versions/#unstable-builds) and
...
+  that is no longer relevant.
+
 - Fix planning of repeat constant expressions in a `GROUP BY` clause
   {{% gh 8302 %}}.
 
+- Support calling multiple distinct [table functions](/sql/functions/#table-func)
+  in the `SELECT` list, as long as those table functions are not nested inside
+  other table functions {{% gh 9988 %}}.
+
 - Add the [`date_bin_hopping`](/sql/functions/date-bin-hopping) function, which
```
[the remark this answers is not in the document]
Comment (sploiselle) [reply doc/user/content/release-notes.md:144]: Oh, that's a good point. I was simply following muscle memory of adding a release note for my work. I agree this should be removed.

[the remark this answers is not in the document]
Comment (sploiselle) [reply doc/user/content/release-notes.md]: Removed in last commit

File: doc/user/content/release-notes.md [hunk 2]
```diff
@@ -107,35 +107,46 @@ These changes are present in [unstable builds](/versions/#unstable-builds) and
...
-
-- Add the `array_cat` function.
-
-{{< comment >}}
-Only add new release notes above this line.
-
-The presence of this comment ensures that PRs that are alive across a release
-boundary don't silently merge their release notes into the wrong place.
-{{</ comment >}}
+- Add experimental support for the [`array_cat`](/sql/functions/#array-func)
```
[the remark this answers is not in the document]
Comment (sploiselle) [reply doc/user/content/release-notes.md]: Removed in last commit

**Comment by benesch** [comment]
Perfect. Good spot @morsapaes and thanks @sploiselle!

**Pull request merged**